odl/operator/operator.py

```diff
@@ -707,6 +707,54 @@ def __call__(self, x, out=None, **kwargs):
                     raise_from(new_exc, err)
         return out
 
+    def norm(self, estimate=False, **kwargs):
```

Comment:
For that case I think you can just steer via `kwargs` how many iterations you run and then add the safety margin yourself.
And yes, keeping things immutable as far as possible is favorable, also in the light of serialization (#1028).